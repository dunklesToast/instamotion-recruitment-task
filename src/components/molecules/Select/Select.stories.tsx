import React from 'react';
import type { Meta, Story } from '@storybook/react';
import type { SelectProps } from './Select';
import { Select } from './Select';

export default {
  title: 'Molecules/Select',
  component: Select,
} as Meta;

const SelectComponent: Story<SelectProps> = (args) => <Select {...args} />;

export const Default = SelectComponent.bind({});
Default.args = {
  options: ['a', 'b'],
  name: 'Buchstaben',
  initField: true,
};
