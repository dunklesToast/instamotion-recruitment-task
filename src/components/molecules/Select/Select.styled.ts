import styled from 'styled-components';

export const Select = styled.select`
  border: 1px black solid;
  border-radius: 4px;
  font-family: 'PT Sans', sans-serif;
  margin-top: 4px;
  padding: 4px;
`;
