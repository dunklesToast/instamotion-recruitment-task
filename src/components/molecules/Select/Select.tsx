import type { ChangeEventHandler } from 'react';
import React from 'react';
import { Text } from '@atoms/Text/Text';
import * as S from './Select.styled';

export interface SelectProps {
  /** All options that should be rendered */
  options: string[];
  /** The Selectbox Name */
  name: string;
  /** The Localized Name which will be used in rendering */
  localizedName: string;
  /** Shows a optional "Choose <localizedName>" Field */
  initField?: boolean;
  /** onChange Listener */
  onChange: ChangeEventHandler<HTMLSelectElement>;
}

/**
 * A simple selectbox
 */
export function Select({
  options,
  name,
  initField,
  onChange,
  localizedName,
}: SelectProps): JSX.Element | null {
  return (
    <>
      <Text weight={700}>{localizedName}</Text>
      <S.Select onChange={onChange} name={name} data-testid='select'>
        {initField ? <option value=''>{localizedName} ausw√§hlen</option> : null}
        {options.map((option, key) => {
          if (option === '') {
            return null;
          }

          return (
            <option key={key} value={option}>
              {option}
            </option>
          );
        })}
      </S.Select>
    </>
  );
}
