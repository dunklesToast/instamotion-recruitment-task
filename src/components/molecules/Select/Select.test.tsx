import React from 'react';
import { render, screen } from '@testing-library/react';
import { Select } from './Select';

test('Select', () => {
  render(<Select onChange={jest.fn} localizedName='Test' name='test' options={['a', 'b']} />);
  expect(screen.getByTestId('select')).toBeInTheDocument();
  expect(screen.getByTestId('select')).toMatchSnapshot();
});
