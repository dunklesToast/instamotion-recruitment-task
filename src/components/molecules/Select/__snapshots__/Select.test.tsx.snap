// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select 1`] = `
.c0 {
  border: 1px black solid;
  border-radius: 4px;
  font-family: 'PT Sans',sans-serif;
  margin-top: 4px;
  padding: 4px;
}

<select
  class="c0"
  data-testid="select"
  name="test"
>
  <option
    value="a"
  >
    a
  </option>
  <option
    value="b"
  >
    b
  </option>
</select>
`;
